ProductManagement/Repositories/ProductRepository.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System.Linq;
+using System.Threading.Tasks;
+using DataAccess.Entities;
+using Microsoft.EntityFrameworkCore;
+
+namespace ProductManagement.Repositories
+{
+    public class ProductRepository : GenericRepository<Product>, IProductRepository
+    {
+        private readonly DbContext _context;
+        
+        private readonly DbSet<Product> _dbSet;
+        
+        public ProductRepository(
+            DbContext context, 
+            DbSet<Product> dbSet) : base(context, dbSet)
+        {
+            _context = context;
+            _dbSet = dbSet;
+        }
+        
+        public Task<Product[]> GetAllAvailableProducts()
+        {
+            var products = _context
+                .Entry(_dbSet)
+                .Entity 
+                .Where(x => x.Accessibility == true)
+                .ToArray();
```

Comment:
```suggestion
                .ToArrayAsync();
```
может еще as-no-tracking пометить?
ниже аналогично